pkg/controller/storagecluster/platform_detection.go

```diff
@@ -45,29 +40,25 @@ func (p *CloudPlatform) GetPlatform(c client.Client) (CloudPlatformType, error)
 	return p.getPlatform(c)
 }
 
-func (p *CloudPlatform) getPlatform(c client.Client) (CloudPlatformType, error) {
-	nodeList := &corev1.NodeList{}
-	err := c.List(context.TODO(), nodeList)
+func (p *Platform) getPlatform(c client.Client) (configv1.PlatformType, error) {
+ infrastructure:=&configv1.Infrastructure{ObjectMeta:metav1.ObjectMeta{Name: "cluster"}}
+	err := c.Get(context.TODO(),types.NamespacedName{Name: infrastructure.ObjectMeta.Name}, infrastructure)
 	if err != nil {
-		return "", fmt.Errorf("could not get storage nodes to determine cloud platform: %v", err)
+		return "", fmt.Errorf("could not get infrastructure details to determine cloud platform: %v", err)
 	}
-	for _, node := range nodeList.Items {
-		providerID := node.Spec.ProviderID
-		for _, cp := range ValidCloudPlatforms {
-			prefix := fmt.Sprintf("%s://", cp)
-			if strings.HasPrefix(providerID, prefix) {
-				p.platform = cp
-				return p.platform, nil
-			}
-		}
-	}
-	p.platform = PlatformUnknown
+
+ infrastructureStatusPlatform:=infrastructure.Status.Platform
+	if(shouldSkipCephObjectDeployment(infrastructureStatusPlatform)){
+			p.platform=infrastructureStatusPlatform
+		 		return p.platform,nil
+		} 
+	p.platform = configv1.NonePlatformType
 	return p.platform, nil
 }
 
-func isValidCloudPlatform(p CloudPlatformType) bool {
-	for _, cp := range ValidCloudPlatforms {
-		if p == cp {
+func shouldSkipCephObjectDeployment(p configv1.PlatformType) bool {
+	for _, platform := range ValidPlatforms {
```

Comment:
Make two new lists `OnPremPlatforms` with baremetal, oVirt, vSphere, etc and `PublicCloudPlatforms` with azure,gcp,aws.

Then use PublicCloudPlatfroms instead of ValidPlatforms here.

Reply:
It makes some sense to also name `shouldSkipCephObjectDeployment` `isPublicCloudPlatform` because that is what we are actually checking.

(I suggested `shouldSkipCephObjectDeployment` in the issue, but now I think `isPublicCloudPlatform` is better)